Elevel/Application/Features/TestCommands/CreateTestCommand.cs

```diff
@@ -156,49 +162,122 @@ private void CreateTestQuestionsForGrammar(List<Guid> questionIds, Test test)
             /// </summary>
             /// <param name="test"></param>
             /// <param name="cancelationtoken"></param>
             /// <returns>true if not created, false if created</returns>
-            private async Task<bool> CreateTestGrammarQuestionsAsync(Test test, CancellationToken cancelationtoken)
+            private async Task<List<TestQuestion>> CreateTestGrammarQuestionsAsync(Test test, CancellationToken cancelationtoken)
             {
                 var questions = await GetQuestionListAsync(test.Level);
                 if (questions.Count() < GRAMMAR_TEST_COUNT)
                 {
-                    return true;
+                    throw new ValidationException("Not Enough questions");
                 }
 
                 var questionIds = GetQuestionIds(questions, GRAMMAR_TEST_COUNT);
 
-                CreateTestQuestionsForGrammar(questionIds, test);
+                var testQuestions = CreateTestQuestions(questionIds, test);
 
-                return false;
+                return testQuestions;
             }
             /// <summary>
             /// Generates Audition test questions for received test
             /// </summary>
             /// <param name="test"></param>
             /// <param name="cancelationtoken"></param>
             /// <returns></returns>
-            private async Task<bool> CreateTestAuditionQuestionsAsync(Test test, CancellationToken cancelationtoken)
+            private async Task<List<TestQuestion>> CreateTestAuditionQuestionsAsync(Test test, CancellationToken cancelationtoken)
             {
                 var questions = await GetQuestionListAsync(test.Level, test.AuditionId);
 
                 if (questions.Count() < AUDITION_TEST_COUNT)
                 {
-                    return true;
+                    throw new ValidationException("Not Enough questions");
                 }
 
                 var questionIds = GetQuestionIds(questions, AUDITION_TEST_COUNT);
 
-                CreateTestQuestionsForGrammar(questionIds, test);
+                var testQuestions = CreateTestQuestions(questionIds, test);
+
+                return testQuestions;
+            }
+
+            private async Task<IEnumerable<Question>> GetQuestionsByAuditionIdAsync(IEnumerable<TestQuestion> testQuestions, Guid? auditionId)
+            {
+                var fileteredQuestions = new List<Question>();
+                var questionsList = await _context.Questions.AsNoTracking().ToListAsync().ConfigureAwait(false);
+                foreach (var testQuestion in testQuestions)
+                {
+                    var question = questionsList.FirstOrDefault(x => x.Id == testQuestion.QuestionId);
+                    if (question.AuditionId == auditionId)
+                    {
+                        fileteredQuestions.Add(question);
+                    }
+                }
+                return fileteredQuestions;
```

Comment:
In this method you can do something like this:
`
var testQuestionIds = testQuestions.Select(x => x.QuestionId);
return await _context.Questions.AsNoTracking().Where(x => x.AuditionId == auditionId && testQuestionIds.Contains(x.Id)).ToListAsync().ConfigureAwait(false);
`